internal/kafka/kafka.go

```diff
@@ -171,11 +172,24 @@ func (b *kafkaBroker) Consume(ctx context.Context, topic string, handler func(ct
 			logger.Warn().Err(err).Msgf("Error when reading message: %s", err.Error())
 		} else {
 			logger.Trace().Bytes("payload", msg.Value).Msgf("Received message with key: %s", msg.Key)
+			ctx, err = ctxval.WithIdentityFrom64(ctx, returnHeaderValue(ctx, msg.Headers))
+			if err != nil {
+				logger.Trace().Msgf("Could not add identity to context: %s", err)
+			}
 			handler(ctx, NewMessageFromKafka(&msg))
 		}
 	}
 }
 
+func returnHeaderValue(ctx context.Context, headers []protocol.Header) string {
+	for _, h := range headers {
+		if h.Key == "x-rh-identity" {
```

Comment:
Headers should be case insensitive, please do:

```suggestion
		if strings.EqualFold(h.Key, "X-RH-Identity") {
```

Or whatever style for HTTP/Kafka headers you prefer.

Reply:
One comment - there is an inconsistency between Kafka native message headers (slice of struct) and our Generic message headers (map). I suggest to change our API to reflect Kafka's because very likely for smaller amount of headers, a list (slice, linked list) is more performant than hash table.

Created: https://issues.redhat.com/browse/HMSPROV-376

Reply:
Should we do that as a follow-up?

Reply:
Yeah, the second comment is meant as a follow-up I believe :)

Reply:
Absolutely, let's not make it more complex, I actually created a ticket for that. This is flaw in my design of GenericMessage, I can do it as well if you don't want to. Let me know.